### PR TITLE
chore: Target cocoapod Sentry/HybridSDK subspec

### LINKED ISSUE #1124: Swift Compiler Error (Xcode): Cannot find 'PrivateSentrySDKOnly' in scope
_Platform:_
- [ ]  Dart
- [x]  Flutter Android or iOS
- [ ]  Flutter Web

_IDE:_
- [x]  VSCode
- [ ]  IntelliJ/AS
- [x]  XCode
- [ ]  Other, which one?

_split-debug-info and obfuscate (Flutter Android or iOS) or CanvasKit (Flutter Web):_
- [ ]  Enabled
- [x]  Disabled

_Platform installed with:_
- [x] pub.dev
- [ ] GitHub

Output of the command `flutter doctor -v` below:

<details>
  <summary>Click me</summary>
  
  ```bash
  Result bundle written to path:
                /var/folders/9k/jkrgwx8d7tg2jg5yqhw2xsk00000gp/T/flutter_tools.jChKIC/flutter_ios_build_temp_dir8cELbH/temporary_xcresult_bundle
           
           
[  +53 ms] Swift Compiler Error (Xcode): Cannot find 'PrivateSentrySDKOnly' in scope
           /Users/mingsern/Development/flutter/.pub-cache/hosted/pub.dartlang.org/sentry_flutter-6.15.0/ios/Classes/SentryFlutterPluginApple.swift:163:39
           
[        ] Swift Compiler Error (Xcode): Cannot find 'PrivateSentrySDKOnly' in scope
           /Users/mingsern/Development/flutter/.pub-cache/hosted/pub.dartlang.org/sentry_flutter-6.15.0/ios/Classes/SentryFlutterPluginApple.swift:166:34
           
[        ] Swift Compiler Error (Xcode): Cannot find 'PrivateSentrySDKOnly' in scope
           /Users/mingsern/Development/flutter/.pub-cache/hosted/pub.dartlang.org/sentry_flutter-6.15.0/ios/Classes/SentryFlutterPluginApple.swift:172:43
           
[        ] Swift Compiler Error (Xcode): Cannot find 'PrivateSentrySDKOnly' in scope
           /Users/mingsern/Development/flutter/.pub-cache/hosted/pub.dartlang.org/sentry_flutter-6.15.0/ios/Classes/SentryFlutterPluginApple.swift:180:24
           
[        ] Swift Compiler Error (Xcode): Cannot find 'PrivateSentrySDKOnly' in scope
           /Users/mingsern/Development/flutter/.pub-cache/hosted/pub.dartlang.org/sentry_flutter-6.15.0/ios/Classes/SentryFlutterPluginApple.swift:195:16
           
[        ] Swift Compiler Error (Xcode): Cannot find 'PrivateSentrySDKOnly' in scope
           /Users/mingsern/Development/flutter/.pub-cache/hosted/pub.dartlang.org/sentry_flutter-6.15.0/ios/Classes/SentryFlutterPluginApple.swift:197:16
           
[        ] Swift Compiler Error (Xcode): Cannot find 'PrivateSentrySDKOnly' in scope
           /Users/mingsern/Development/flutter/.pub-cache/hosted/pub.dartlang.org/sentry_flutter-6.15.0/ios/Classes/SentryFlutterPluginApple.swift:229:13
           
[        ] Swift Compiler Error (Xcode): Cannot find 'PrivateSentrySDKOnly' in scope
           /Users/mingsern/Development/flutter/.pub-cache/hosted/pub.dartlang.org/sentry_flutter-6.15.0/ios/Classes/SentryFlutterPluginApple.swift:230:13
           
[        ] Swift Compiler Error (Xcode): Cannot find 'PrivateSentrySDKOnly' in scope
           /Users/mingsern/Development/flutter/.pub-cache/hosted/pub.dartlang.org/sentry_flutter-6.15.0/ios/Classes/SentryFlutterPluginApple.swift:384:29
           
[        ] Swift Compiler Error (Xcode): Cannot find 'PrivateSentrySDKOnly' in scope
           /Users/mingsern/Development/flutter/.pub-cache/hosted/pub.dartlang.org/sentry_flutter-6.15.0/ios/Classes/SentryFlutterPluginApple.swift:389:8
           
[        ] Swift Compiler Error (Xcode): Cannot find 'PrivateSentrySDKOnly' in scope
           /Users/mingsern/Development/flutter/.pub-cache/hosted/pub.dartlang.org/sentry_flutter-6.15.0/ios/Classes/SentryFlutterPluginApple.swift:395:40
           
[        ] Swift Compiler Error (Xcode): Cannot find 'PrivateSentrySDKOnly' in scope
           /Users/mingsern/Development/flutter/.pub-cache/hosted/pub.dartlang.org/sentry_flutter-6.15.0/ios/Classes/SentryFlutterPluginApple.swift:418:12
           
[        ] Swift Compiler Error (Xcode): Cannot find 'PrivateSentrySDKOnly' in scope
           /Users/mingsern/Development/flutter/.pub-cache/hosted/pub.dartlang.org/sentry_flutter-6.15.0/ios/Classes/SentryFlutterPluginApple.swift:424:26
           
[        ] Swift Compiler Error (Xcode): Cannot find 'PrivateSentrySDKOnly' in scope
           /Users/mingsern/Development/flutter/.pub-cache/hosted/pub.dartlang.org/sentry_flutter-6.15.0/ios/Classes/SentryFlutterPluginApple.swift:437:12
           
[        ] Swift Compiler Error (Xcode): Cannot find 'PrivateSentrySDKOnly' in scope
           /Users/mingsern/Development/flutter/.pub-cache/hosted/pub.dartlang.org/sentry_flutter-6.15.0/ios/Classes/SentryFlutterPluginApple.swift:443:26
           
[   +1 ms] Could not build the application for the simulator.
[        ] Error launching application on iPhone 14 Pro Max.
[   +1 ms] "flutter run" took 24,349ms.
[   +1 ms] 
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
           #1      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:713:9)
           <asynchronous suspension>
           #2      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1209:27)
           <asynchronous suspension>
           #3      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #4      CommandRunner.runCommand (package:args/command_runner.dart:209:13)
           <asynchronous suspension>
           #5      FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:281:9)
           <asynchronous suspension>
           #6      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #7      FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:229:5)
           <asynchronous suspension>
           #8      run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:62:9)
           <asynchronous suspension>
           #9      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #10     main (package:flutter_tools/executable.dart:91:3)
           <asynchronous suspension>
           
           
[        ] Running shutdown hooks
[        ] Shutdown hooks complete
[        ] exiting with code 1
  ```
</details>

The version of the SDK (See pubspec.lock):
**6.15.0**

---
I have the following issue:

Running `flutter run --flavor=development` on iOS 16, iPhone 14 Pro Max simulator, this error pops up.
Already tried flutter pub get, clean, pod install, update, cache clean, cache repair, seems like none of it works.

**Steps to reproduce:**
- run flutter run --flavor=development

**Actual result:**
- Unable to run flutter app

**Expected result:**
- Able to run flutter app without any problem

### PR DESCRIPTION
**Pull request by brustolin**
Add a dependency to Sentry/HybridSDK cocoa subspec, to have access to `PrivateSentrySDKOnly.h`.

We need to release the next iOS version before merging this.

close #1124

_#skip-changelog_

### CODE REVIEW

**Comment by krystofwoldrich** [comment]
@brustolin What version will it be in v7.32.0?

File: flutter/ios/sentry_flutter.podspec [hunk 1]
```diff
@@ -12,7 +12,7 @@ Sentry SDK for Flutter with support to native through sentry-cocoa.
                          :tag => s.version.to_s }
   s.source_files     = 'Classes/**/*'
   s.public_header_files = 'Classes/**/*.h'
-  s.dependency 'Sentry', '~> 7.30.1'
+  s.dependency 'Sentry/HybridSDK', '~> 7.31.0'
```
Comment (brustolin) [inline flutter/ios/sentry_flutter.podspec]:
```suggestion
  s.dependency 'Sentry/HybridSDK', '~> 7.30.2'
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -13,9 +13,9 @@
 - Bump Android SDK from v6.6.0 to v6.7.0 ([#1105](https://github.com/getsentry/sentry-dart/pull/1105))
   - [changelog](https://github.com/getsentry/sentry-java/blob/main/CHANGELOG.md#670)
   - [diff](https://github.com/getsentry/sentry-java/compare/6.6.0...6.7.0)
-- Bump Cocoa SDK from v7.30.0 to v7.30.1 ([#1104](https://github.com/getsentry/sentry-dart/pull/1104))
-  - [changelog](https://github.com/getsentry/sentry-cocoa/blob/master/CHANGELOG.md#7301)
-  - [diff](https://github.com/getsentry/sentry-cocoa/compare/7.30.0...7.30.1)
+- Bump Cocoa SDK from v7.30.0 to v7.31.0 ([#1104](https://github.com/getsentry/sentry-dart/pull/1113))
+  - [changelog](https://github.com/getsentry/sentry-cocoa/blob/master/CHANGELOG.md#7310)
+  - [diff](https://github.com/getsentry/sentry-cocoa/compare/7.30.0...7.31.0)
```
Comment (marandaneto) [inline CHANGELOG.md]: @brustolin changelog is outdated now.

**Comment by marandaneto** [comment]
We can close https://github.com/getsentry/sentry-dart/pull/1123 if this is green.

File: CHANGELOG.md [hunk 3]
```diff
@@ -13,9 +13,9 @@
 - Bump Android SDK from v6.6.0 to v6.7.0 ([#1105](https://github.com/getsentry/sentry-dart/pull/1105))
   - [changelog](https://github.com/getsentry/sentry-java/blob/main/CHANGELOG.md#670)
   - [diff](https://github.com/getsentry/sentry-java/compare/6.6.0...6.7.0)
-- Bump Cocoa SDK from v7.30.0 to v7.30.1 ([#1104](https://github.com/getsentry/sentry-dart/pull/1104))
-  - [changelog](https://github.com/getsentry/sentry-cocoa/blob/master/CHANGELOG.md#7301)
-  - [diff](https://github.com/getsentry/sentry-cocoa/compare/7.30.0...7.30.1)
+- Bump Cocoa SDK from v7.30.0 to v7.30.2 ([#1113](https://github.com/getsentry/sentry-dart/pull/1113))
+   - [changelog](https://github.com/getsentry/sentry-cocoa/blob/master/CHANGELOG.md#7302)
+   - [diff](https://github.com/getsentry/sentry-cocoa/compare/7.30.0...7.30.2)
```
Comment (marandaneto) [inline CHANGELOG.md]:
```suggestion
- Bump Cocoa SDK from v7.30.0 to v7.30.2 ([#1113](https://github.com/getsentry/sentry-dart/pull/1113))
  - [changelog](https://github.com/getsentry/sentry-cocoa/blob/master/CHANGELOG.md#7302)
  - [diff](https://github.com/getsentry/sentry-cocoa/compare/7.30.0...7.30.2)
```

**Review by marandaneto — APPROVED** [review]

**Pull request merged**